simple-acquaintances/data/datagen.py

```diff
@@ -0,0 +1,269 @@
+            '{} is not a non negative integer.'.format(arg))
+    return int(arg)
+
+
+def parse_args():
+    """Parse CLI arguments."""
+    parser = argparse.ArgumentParser(
+        description='Generate data for the PSL simple-acquaintances example.')
+    parser.add_argument('--people', default=25, type=checkPositiveInt,
+                        help='Number of people in your dataset.')
+    parser.add_argument(
+        '--places',
+        default=5,
+        type=checkPositiveInt,
+        help='Number of places in your dataset.')
+    parser.add_argument(
+        '--placesLivedMean',
+        default=3,
+        type=checkPositiveInt,
+        help='Mean of the number of places a person has lived in.')
+    parser.add_argument(
+        '--placesLivedSD',
+        default=2,
+        type=checkPositiveInt,
+        help='Standard deviation of the number of places a person has lived in.')
+    parser.add_argument('--globalLikes', default=5, type=checkPositiveInt,
+                        help='Number of global things to like.')
+
+    parser.add_argument(
+        '--localLikes',
+        default=5,
+        type=checkPositiveInt,
+        help='Number of local things to like.')
+    parser.add_argument(
+        '--localLikesVariance',
+        default=1,
+        type=checkPositiveInt,
+        help='Variance for likability of a local thing based on place lived.')
+    args = parser.parse_args()
+    return args
+
+
+def get_truncated_normal(mean=0, sd=1, low=0, upp=10):
+    """Return truncated normal distribution."""
+    return truncnorm(
+        (low - mean) / sd, (upp - mean) / sd, loc=mean, scale=sd)
+
+
+class Person():
+    """Person class for the dataset."""
+
+    def __init__(self):
+        self.index = None
+        self.lived = None
+        self.globalLikes = None
+        self.localLikes = None
+
+
+class DataGen():
+    """Generate data for simple-acquaintances example."""
+
+    def __init__(
+            self,
+            noOfPeople,
+            noOfPlaces,
+            noOfGlobal,
+            noOfLocal,
+            placesLivedMean,
+            placesLivedSD,
+            localLikesVariance):
+        self.noOfPeople = noOfPeople
+        self.places = np.arange(noOfPlaces)
+        self.globalThings = np.arange(noOfGlobal)
+        self.localThings = np.arange(noOfLocal)
+
+        self.placesAffectingLocalThings = {}
+        self.__generatePALT()  # fill in placesAffectingLocalThings
+
+        self.placesLivedMean = placesLivedMean
+        self.placesLivedSD = placesLivedSD
+        self.placesLivedUpper = noOfPlaces
+        self.localLikesVariance = localLikesVariance
+        # self.truncnormGenerator = get_truncated_normal(mean, sd, 1, upper)
+
+    def __generatePALT(self):
+        """Generate porbabilities of places affecting the likable-ity of local things."""
+        # eg: [Place1:[LThing1, Lthing2], Place2:[LThing1,LThing2]]
+
+        noOfLocalThings = self.localThings.size
```

Comment:
`len()` provides number of rows in case of 2D matrices. However `.size` always returns the number of elements in the numpy array. Hence I would like to keep it as it is.

Reply:
If self.localThings was a 2D matrix, then this code will break because of the later index into it.
So that point doesn't hold any water.
Unless you have a stronger concern, keep with idiomatic python.

Also we also discussed the possibility of reducing these down to simpler data types.
In which case, this would be moot.